armi/reactor/converters/uniformMesh.py

```diff
@@ -102,23 +106,66 @@ def __init__(self, cs=None):
         # This prevents clearing out data on the original reactor that should
         # be preserved since no changes were applied.
         self._cachedReactorCoreParamData = {}
+        self._nonUniformMeshFlags = [
+            Flags.fromStringIgnoreErrors(f) for f in cs["nonUniformAssemFlags"]
+        ]
+        self._hasNonUniformAssems = any(self._nonUniformMeshFlags)
 
     def convert(self, r=None):
         """Create a new reactor core with a uniform mesh."""
         if r is None:
             raise ValueError(f"No reactor provided in {self}")
 
-        runLog.extra("Building copy of {} with a uniform axial mesh".format(r))
+        runLog.extra(f"Building with a uniform axial mesh reactor from {r}")
         completeStartTime = timer()
         self._sourceReactor = r
-        self.convReactor = self.initNewReactor(r)
-        self._setParamsToUpdate()
-        self._computeAverageAxialMesh()
-        self._buildAllUniformAssemblies()
-        self._mapStateFromReactorToOther(self._sourceReactor, self.convReactor)
-        self.convReactor.core.updateAxialMesh()
 
-        self._newAssembliesAdded = self.convReactor.core.getAssemblies()
+        # Here we are taking a short cut to homogenizing the core by only focusing on the
+        # core assemblies that need to be homogenized. This will have a large speed up
+        # since we don't have to create an entirely new reactor perform the data mapping.
+        # We are using the Spent Fuel Pool (SFP) as a container just to temporarily store
```

Comment:
So I get that in this context the SFP serves as a temporary container and assemblies that get added to it here are soon removed down in `_mapStateFromReactorToOther`. But it feels like this could be error prone in the long term. Moving assemblies to a data structure that can persist outside of this class feels.... fishy. Especially when it's filling a short term/temporary local namespace storage need.

I wonder if it would be safer to create an instance variable (e.g., `self.tempAssemblyStorage = []` or something) that we know will go out of scope when the class instance is finished up. If/when we tackle #876 we may have to make this change anyway. 

Just something to consider.

Reply:
Got it! Updated.

Reply:
Awesome! Can you update the comment too? The big explanatory one right before `if self._hasNonUniformAssems`.

Reply:
https://github.com/terrapower/armi/pull/879/commits/ac6bc4324de00c0029fb9dea4fdde121e4ad7585
https://github.com/terrapower/armi/pull/879/commits/839885e81835c9a9d2d39d6db393234ddedf00e9
https://github.com/terrapower/armi/pull/879/commits/f84a5b865fe5f7e6088853cc44964b07c60aeb88